crates/driver/src/domain/competition/auction.rs

```diff
@@ -209,8 +231,45 @@ impl AuctionProcessor {
         fut
     }
 
-    /// Removes orders that cannot be filled due to missing funds of the owner.
-    fn filter_orders(balances: &mut Balances, orders: &mut Vec<order::Order>) {
+    /// Fetches the app data for all orders in the auction.
+    /// Returns a map from order UIDs to the result of fetching the app data.
+    async fn collect_orders_app_data(
+        app_data_retriever: Option<order::app_data::AppDataRetriever>,
+        orders: &[order::Order],
+    ) -> HashMap<order::Uid, Option<app_data::ValidatedAppData>> {
```

Comment:
Not sure this is the best return type:
1. I believe we don't need to have the value optional. If we want to express no appdata could be fetched we could just omit the entry from the map.
2. As you pointed out the majority of appdata values is duplicated so why don't we return a mapping `hash => ValidatedAppData` instead?